### DISCUSSION TITLE
Move to text channel

### DISCUSSION THREAD

**Discussion by whyrmph** [Rapptz/discord.py#7883] [Q&A]
Hello, please tell me, I have such a question.
Is it possible to **move** to another text chat?

That is , let 's say: 
I have **two roles**, they are responsible for their categories and chats in them.
I implemented the move by clicking on the button.
The button inherently **removes** the role in which the first category is located, and **adds** a role with the second category.
But at the same time, when changing roles, **I remain in the chat with the first category**.
How to implement, when clicking on the button, moving from the chat of the first category to the chat of the second category?

I tried using `discord.TextChannel.move`, but it didn't give any result.

_Sorry if I didn't ask clearly, I use Google translator._

**Comment by LostLuma** [comment]
You can not change which channel the user's client is focused on.

A solution you might want to consider is to provide the user with a [channel mention](https://discordpy.readthedocs.io/en/latest/api.html#discord.abc.GuildChannel.mention) or [jump link](https://discordpy.readthedocs.io/en/latest/api.html#discord.abc.GuildChannel.jump_url) to make the channel switch more obvious. You would need to send the response before changing the roles if the channel is only visible for users with the role, else the user will not be able to read the message.

The `TextChannel.move` method changes the channel's position in the channel list on the left, just like dragging it with your mouse does, it is indeed unrelated.

Note that there's also the [official discord.py server](https://discord.gg/r3sSKJJ) where you can ask questions, for help with the alpha version use the `?tester` command and ask in the `#beta-testing` channel.
